pkg/cmd/roachtest/roachtestutil/utils.go

```diff
@@ -118,10 +125,27 @@ func CreateStatsFileInClusterFromExporter(
 	exporter exporter.Exporter,
 	node option.NodeListOption,
 ) (string, error) {
+	return CreateStatsFileInClusterFromExporterWithPrefix(ctx, t, c, perfBuf, exporter, node, "")
+}
+
+func CreateStatsFileInClusterFromExporterWithPrefix(
+	ctx context.Context,
+	t test.Test,
+	c cluster.Cluster,
+	perfBuf *bytes.Buffer,
+	exporter exporter.Exporter,
+	node option.NodeListOption,
+	prefix string,
+) (string, error) {
+
+	if perfBuf == nil {
+		return "", errors.New("perf buffer is nil")
+	}
+	// Close and flush any existing data in the buffer
 	if err := exporter.Close(nil); err != nil {
```

Comment:
Seems weird to do it in the callee as opposed to the caller. The caller should take care of closing the exporter, thus ensuring that the output buffer is "whole". This is just plain old separation of concerns. Thus, the callee need not be aware of the `exporter`.

Reply:
This is something that I'm fixing in this PR https://github.com/cockroachdb/cockroach/pull/135821.

If I remove it in this PR, the other roachtests will fail. One way is that I merge the above mentioned PR into this PR, but I wanted this PR to be only for tpccbench

Reply:
I think if you rebase this PR on top of https://github.com/cockroachdb/cockroach/pull/135821, GH will hide base changes, or at least Reviewable will. It would make it easier to review the _union_ of the changes.

Reply:
Makes sense.
Let me include the changes in this changes itself.

Reply:
I have merged the PRs into this one.